src/legacy/core_plugins/kibana/public/dashboard/migrations/migrate_to_730_panels.ts

```diff
@@ -0,0 +1,304 @@
+ * same data.
+ */
+
+function isPre61Panel(
+  panel: unknown | RawSavedDashboardPanelTo60
+): panel is RawSavedDashboardPanelTo60 {
+  return (panel as RawSavedDashboardPanelTo60).row !== undefined;
+}
+
+function is61Panel(panel: unknown | RawSavedDashboardPanel610): panel is RawSavedDashboardPanel610 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel610).version);
+  return panelVersion && panelVersion.major === 6 && panelVersion.minor === 1;
+}
+
+function is62Panel(panel: unknown | RawSavedDashboardPanel620): panel is RawSavedDashboardPanel620 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel620).version);
+  return panelVersion && panelVersion.major === 6 && panelVersion.minor === 2;
+}
+
+function is63Panel(panel: unknown | RawSavedDashboardPanel630): panel is RawSavedDashboardPanel630 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel630).version);
+  return panelVersion && panelVersion.major === 6 && panelVersion.minor === 3;
+}
+
+function is640To720Panel(
+  panel: unknown | RawSavedDashboardPanel640To720
+): panel is RawSavedDashboardPanel640To720 {
+  const panelVersion = PanelUtils.parseVersion((panel as RawSavedDashboardPanel640To720).version);
+  return (
+    panelVersion &&
+    ((panelVersion.major === 6 && panelVersion.minor > 3) ||
+      (panelVersion.major === 7 && panelVersion.minor < 3))
+  );
+}
+
+// Migrations required for 6.0 and prior:
+// 1. (6.1) migrate size_x/y/row/col into gridData
+// 2. (6.2) migrate uiState into embeddableConfig
+// 3. (6.3) scale grid dimensions
+// 4. (6.4) remove columns, sort properties
+// 5. (7.3) make sure panelIndex is a string
+function migratePre61PanelToLatest(
+  panel: RawSavedDashboardPanelTo60,
+  version: string,
+  useMargins: boolean,
+  uiState?: { [key: string]: { [key: string]: unknown } }
+): RawSavedDashboardPanel730ToLatest {
+  ['col', 'row'].forEach(key => {
+    if (!_.has(panel, key)) {
+      throw new Error(
+        i18n.translate('kbn.dashboard.panel.unableToMigratePanelDataForSixOneZeroErrorMessage', {
+          defaultMessage:
+            'Unable to migrate panel data for "6.1.0" backwards compatibility, panel does not contain expected field: {key}',
+          values: { key },
+        })
+      );
+    }
+  });
+
+  const embeddableConfig = uiState ? uiState[`P-${panel.panelIndex}`] || {} : {};
+
+  if (panel.columns || panel.sort) {
+    embeddableConfig.columns = panel.columns;
+    embeddableConfig.sort = panel.sort;
+  }
+
+  const heightScaleFactor = useMargins
+    ? PANEL_HEIGHT_SCALE_FACTOR_WITH_MARGINS
+    : PANEL_HEIGHT_SCALE_FACTOR;
+
+  const { columns, sort, row, col, size_x, size_y, ...rest } = panel;
+  return {
+    ...rest,
+    version,
+    panelIndex: panel.panelIndex.toString(),
+    gridData: {
+      x: (panel.col - 1) * PANEL_WIDTH_SCALE_FACTOR,
```

Comment:
panel.col could just be col, since you destructured it on line 123. Same goes for panel.row, panel.size_x, etc below.